cpp/src/arrow/filesystem/azurefs_test.cc

```diff
@@ -507,6 +507,41 @@ TEST_F(AzuriteFileSystemTest, CreateDirUri) {
   ASSERT_RAISES(Invalid, fs_->CreateDir("abfs://" + RandomContainerName(), true));
 }
 
+TEST_F(AzuriteFileSystemTest, DeleteContainerDirSuccess) {
+  auto container_name = RandomContainerName();
+  ASSERT_OK(fs_->CreateDir(container_name));
+  ASSERT_OK(fs_->DeleteDir(container_name));
+}
+
+TEST_F(AzuriteFileSystemTest, DeleteDirSuccess) {
+  const auto path =
+      internal::ConcatAbstractPath(PreexistingContainerName(), RandomDirectoryName());
+  // There is only virtual directory without hierarchical namespace
+  // support. So the DeleteDir() does nothing.
```

Comment:
Shouldn't it delete all the blobs that start with `path/to/dir/`?

Reply:
It make sense. I'll implement it and add a test for the case.